design/proposals/variable_provision.md

```diff
@@ -0,0 +1,295 @@
+# Solution Design - Variable Provisioning
+
+## Table of Contents
+
+- [Glossary](#glossary)
+- [Useful links](#useful-links)
+- [Background](#background)
+- [Issue description](#issue-description)
+  * ['Browsing-up` for secrets management](#-browsing-up--for-secrets-management)
+  * [Secrets Domain Knowledge Required](#secrets-domain-knowledge-required)
+  * [Human Intervention Required](#human-intervention-required)
+  * [Credential Provenance](#credential-provenance)
+- [Solution](#solution)
+  * [Design](#design)
+  * [Backwards compatibility](#backwards-compatibility)
+  * [Performance](#performance)
+  * [Affected Components](#affected-components)
+- [Security](#security)
+- [Test Plan](#test-plan)
+- [Logs](#logs)
+  * [Audit](#audit)
+- [Documentation](#documentation)
+- [Version update](#version-update)
+- [Open questions](#open-questions)
+- [Breaking](#breaking)
+
+
+## Glossary
+[//]: # (Describe terms that will be used throughout the design)
+[//]: # (You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#)
+
+> TBD
+
+| **Term** | **Description** |
+|----------|-----------------|
+|          |                 |
+|          |                 |
+
+## Useful links
+[//]: # (Add links that may be useful for the reader)
+
+> TBD
+
+## Background
+
+The purpose of Variables in Conjur is to put values in them for applications to
+use for service access. These values often consist of:
+
+- Passwords
+- API Keys
+- x509 Private Keys
+- x509 Certificates
+
+The typical workflow for a Variable is:
```

Comment:
Just a note (mostly to myself), it would be good to visualize what a mess this flow actually looks like.  Particularly with multiple actors (ex. Security engineer to create policy, database admin to set value, application to use the value).